index.bs

```diff
@@ -0,0 +1,221 @@
+    1. Call <a>compute the layout shift</a> with <var>unstable elements</var>.
+</div>
+
+Compute the Layout Shift {#sec-compute-layout-shift}
+-----------------------------------------------------
+
+<div algorithm="compute the layout shift">
+    When asked to to <dfn export>compute the layout shift</dfn>, with <var>unstable elements</var>, a list of impacted <a>element</a>'s on the page, as input, run the following steps:
+
+    1. Let <var>impacted region</var> be initially set to empty.
+    1. Let <var>max shift fraction</var> be initially set to 0.
+    1. For each tuple <var>tuple</var> in <var>unstable elements</var>:
+        1. Take the geometric union of <var>impacted region</var> with <var>tuple</var>’s <a>currentVisualRepresentation</a> and <var>tuple</var>’s <a>previousVisualRepresentation</a> if <a>previousVisualRepresentation</a> is not null.
+        1. If <var>tuple</var>’s <var>shiftFraction</var> > <var>max shift fraction</var>, <var>max shift fraction</var> equals <var>shiftFraction</var>.
+    1. Let <var>impacted fraction</var> be the area of the <var>impacted region</var> divided by the area of the current <a>viewport</a>.
+    1. The <var>layout shift</var> is the <var>impacted fraction</var> multiplied by the <var>max shift fraction</var>.  
+    1. If <var>layout shift</var> is greater than 0, create a new {{LayoutShift}} object |newEntry| and set its attributes as follows:
+        1. Set |newEntry|'s {{PerformanceEntry/name}} attribute to <code>"layout-shift"</code>.
+        1. Set |newEntry|'s {{PerformanceEntry/entryType}} attribute to <code>layoutShift</code>.
+        1. Set |newEntry|'s {{PerformanceEntry/startTime}} attribute to <a>current high resolution time</a>.
+        1. Set |newEntry|'s {{PerformanceEntry/duration}} attribute to 0.
+        1. Set |newEntry|'s {{LayoutShift/value}} attribute to <var>layout shift</var>.
+    1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a> |newEntry| object.
+</div>
+
+NOTE: This computation ensures that the layout shift takes into account both the fraction of the viewport that has been impacted by layout stability as well as the greatest impact to any given element in the viewport.  This is to recognize that a large element that moves only a small distance may have a small impact on the perceived stability of the page.
+
+
+Identify an Unstable Element {#sec-identify-unstable-element}
+--------------------------------------------------------------
+
+<div algorithm="identify an unstable element">
+    When asked to <dfn export>identify an unstable element</dfn>, given <var>element</var> as input, run the following steps:
+
+    1. If <var>element</var>’s <a>previousFrameStartingPoint</a> is null, return false.
+    1. If <a>currentFrameStartingPoint</a> is not at <a>previousFrameStartingPoint</a> and it has changed by more than 3 CSS pixels, return true.
```

Comment:
Feels like we need to link to the definition of a "CSS pixel", though I'm not sure where best to do that. The closest thing I could find is [here](https://www.w3.org/TR/cssom-view-1/#css-pixels), but that also doesn't really define them. Maybe Tab would know?

Reply:
That seems reasonable.  I am not sure who Tab is, but I have added Steve in case he has any reference points.

@skobes

Reply:
https://www.w3.org/TR/css-values-4/#pixel-unit

Reply:
/cc @tabatkins (in case he has thoughts)

Reply:
These "starting points" appear to be defined abstractly as just positions; they should be nailed down a little more concretely as offsets from some particular container (the top of the document?) and in some particular coordinate space (the initial coordinate space?) & unit (CSS px), then you can just do a simple numeric comparison of "differs by 3 or more"

Reply:
Ok I added some things to the definition of starting point in terminology to hopefully make it more clear.  Please look above and see if that is less ambiguous.

Reply:
It looks like you're linking "initial coordinate" to the SVG spec, which I would avoid.  Can we refer instead to the coordinate space of the initial containing block (https://www.w3.org/TR/CSS2/visudet.html#containing-block-details)?

Reply:
done